bids/layout/__init__.py

```diff
@@ -21,3 +23,9 @@
     "Tag",
     "Query"
 ]
+
+try:
+    from .layout_v2 import BIDSLayoutV2
+    __all__ += ["BIDSLayoutV2"]
+except:
+    warnings.warn("Could not load BIDSLayoutV2: make sure you installed the ancpBIDS package")
```

Comment:
We could make it always available, but only raise an error if someone tries to use it:

```suggestion
    "Query",
    "BIDSLayoutV2",
]

try:
    from .layout_v2 import BIDSLayoutV2
except Exception as err:
    def BIDSLayoutV2(*args, **kwargs):
        raise RuntimeError("Cannot create BIDSLayoutV2 - please install the ancpbids package.") from err
```